### PR TITLE
[FAU-414] Add support for `ids` attribute to degree programs overview shortcode

### PR DESCRIPTION
**Pull request by amiut**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)
https://inpsyde.atlassian.net/browse/FAU-414
No support for `ids` attribute in degree programs overview shortcode.

**What is the new behavior (if this is a feature change)?**
Add support for `ids` attribute to degree programs overview shortcode.
the `ids` attribute takes a comma-separated list of HIS codes

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Related PRs**
- https://github.com/RRZE-Webteam/FAU-Studium/pull/132
- https://github.com/RRZE-Webteam/FAU-Studium-Common/pull/7

**Other information**:

### CODE REVIEW

**Review by shvlv — APPROVED** [review]

**Review by zhyian — APPROVED** [review]
Thanks, LGTM and works well locally.

**Review by o-samaras — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Expand the admission requirements filter to include the "Admission free with restriction" option.
 - Support filter reordering based on the `filters` attribute in the search shortcode.
+- Support `ids` attribute in the search shortcode.
```
Comment (tyrann0us) [inline CHANGELOG.md:15]:
```suggestion
- Support Campo Keys in the search shortcode (`ids` attribute).
```

File: docs/degree_programs_search_shortcode.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ See below for a list of supported attributes.
 | `hide`          | Comma-separated names of elements that should be hidden            | no       |         | See list below    |
 | `output`        | Output mode                                                        | no       | `list`  | `list` or `tiles` |
 | `filters`       | Comma-separated list of filters to be shown to visitors            | no       |         | See list below    |
+| `ids`           | Comma-separated list of HIS codes                                  | no       |         | See list below    |
```
Comment (tyrann0us) [inline docs/degree_programs_search_shortcode.md:15]: Even if we name it "HIS codes" in the source code, let's name it "Campo Keys" in all user-facing places.
```suggestion
| `ids`           | Comma-separated list of Campo Keys                                 | no       |         | See list below    |
```

File: docs/degree_programs_search_shortcode.md [hunk 3]
```diff
@@ -68,3 +69,13 @@ slugs. This does **not** work on other websites.
...
+
+### Usage of `ids` attribute
+
+The `ids` attribute takes a comma-separated list of HIS codes, HIS codes can be complete or partial. Valid examples are:
+
+- 08|185|-|-|H|1|E|P|V|1|
+- 08|185
+- 08|185,08|145,08|185|-|-|H|1|E|P|V|1|
+
+currently, only position 1,2 and 7 of the HIS code are used, other values are ignored, in the first example of above, value `08`, `185` and `E` are used, others are ignored.
```
Comment (tyrann0us) [inline docs/degree_programs_search_shortcode.md:81]:
```suggestion
### Campo Keys (`ids` attribute)

Degree programs can be filtered by Campo Keys. To do this, use the `ids` attribute, which takes a comma-separated list of Campo Keys. Note that Campo Keys can be complete or partial. Valid examples are

- `08|185|-|-|H|1|E|P|V|1|`
- `08|185`
- `08|185,08|145,08|185|-|-|H|1|E|P|V|1|`

Currently, only positions 1, 2, and 7 of the Campo Key are used; other values are ignored. In the first example above, only the values `08`, `185`, and `E` are used.
```

**Review by tyrann0us — COMMENTED** [review]
Thanks! Left a couple of wording-related comments. Will do functional testing later.

**Pull request merged**